cpp/src/arrow/flight/client.cc

```diff
@@ -442,8 +549,15 @@ class FlightClient::FlightClientImpl {
       args.SetSslTargetNameOverride(options.override_hostname);
     }
 
+    std::vector<std::unique_ptr<grpc::experimental::ClientInterceptorFactoryInterface>>
+        interceptors;
+    interceptors.emplace_back(
+        new GrpcClientInterceptorAdapterFactory(std::move(options.middleware)));
```

Comment:
We shouldn't call `std::move` on a const options object reference.

Reply:
I changed the method to take the options by const reference, and to pass auth via shared pointer.